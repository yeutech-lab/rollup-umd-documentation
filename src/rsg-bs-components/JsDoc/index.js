export { default } from 'rsg-components/JsDoc';
