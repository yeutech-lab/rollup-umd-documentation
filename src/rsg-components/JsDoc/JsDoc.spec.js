import React from 'react';
import JsDoc, { getMarkdown } from './JsDoc';

const tags = {
	deprecated: [
		{
			title: 'description',
			description: 'Use *another* method',
		},
	],
	version: [
		{
			title: 'version',
			description: '2.0.0',
		},
	],
	since: [
		{
			title: 'since',
			description: '1.0.0',
		},
	],
	author: [
		{
			title: 'author',
			description: '[Author 1](#TestLink)',
		},
		{
			title: 'author',
			description: '[Author 2](#TestLink2)',
		},
	],
	see: [
		{
			title: 'see',
			description: '[See 1](#TestLink)',
		},
		{
			title: 'see',
			description: '[See 2](#TestLink2)',
		},
	],
	link: [
		{
			title: 'link',
			description: '[Link 1](#TestLink)',
		},
	],
};

describe('getMarkdown', () => {
	it('should return Markdown for all tags', () => {
		const result = getMarkdown(tags);
		expect(result).toMatchSnapshot();
	});

	it('should return Markdown for one author', () => {
		const result = getMarkdown({
			author: [tags.author[0]],
		});
		expect(result).toMatchSnapshot();
	});

	it('should return Markdown for multiple authors', () => {
		const result = getMarkdown({
			author: tags.author,
		});
		expect(result).toMatchSnapshot();
	});
});

describe('JsDoc', () => {
	it('should render Markdown', () => {
		const actual = shallow(<JsDoc {...tags} />);

		expect(actual).toMatchSnapshot();
	});

	it('should render null for empty tags', () => {
		const actual = shallow(<JsDoc />);

		expect(actual.getElement()).toBe(null);
	});
});
