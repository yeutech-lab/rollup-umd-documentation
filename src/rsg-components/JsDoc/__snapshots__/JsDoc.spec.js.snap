// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsDoc should render Markdown 1`] = `
<Markdown
  text="**Deprecated:** Use *another* method

[See 1](#TestLink)

[See 2](#TestLink2)

[Link 1](#TestLink)

Authors: [Author 1](#TestLink), [Author 2](#TestLink2)

Version: 2.0.0

Since: 1.0.0"
/>
`;

exports[`getMarkdown should return Markdown for all tags 1`] = `
"**Deprecated:** Use *another* method

[See 1](#TestLink)

[See 2](#TestLink2)

[Link 1](#TestLink)

Authors: [Author 1](#TestLink), [Author 2](#TestLink2)

Version: 2.0.0

Since: 1.0.0"
`;

exports[`getMarkdown should return Markdown for multiple authors 1`] = `"Authors: [Author 1](#TestLink), [Author 2](#TestLink2)"`;

exports[`getMarkdown should return Markdown for one author 1`] = `"Author: [Author 1](#TestLink)"`;
